cmd/create/cluster/cmd.go

```diff
@@ -2088,6 +2102,78 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// If OCP version >= 4.10 only
+	var machinePoolRootDisk *ocm.Volume
+
+	// Setting the machine pool root disk size is only supported on gp3 volumes
+	gp3SupportedVersion := "4.10.0"
+
+	// As of 4.10, GP3 is the default whenever possible, so this is sufficient to implement the
+	// disk size on GP3 only
+	// The only exception is when the availability zone is either local or wavelength, however this
+	// situation does not exist since LocalZone is only available as of 4.12
+	// OCM already has the fallback code to use gp2 if gp3 is not supported on a LocalZone
+	isGP3Supported, err := ocm.IsGreaterThanOrEqual(version, gp3SupportedVersion)
+	if err != nil {
+		r.Reporter.Errorf("Failed to check if the OCP version is greater than or equal to %s: %v",
+			gp3SupportedVersion, err)
+		os.Exit(1)
+	}
+
+	// If the user sets the machine pool root disk size and the cluster is not 4.10 or greater, we
+	// print a warning and ignore the value
+	if !isGP3Supported && args.machinePoolRootDiskSize != "" {
+		r.Reporter.Warnf("Setting the machine pool root disk size is only supported on OpenShift " +
+			"4.10 or greater. Ignoring the value.")
+	}
+
+	// Allow non -GP3 volume if the AZ is local or wavelength
+	if isGP3Supported &&
+		(args.machinePoolRootDiskSize != "" || interactive.Enabled()) {
```

Comment:
Can we please extract this new logic to a new function like `setMachinePoolRootDiskSize`? (lines `2132-2176`)

Reply:
I'm concerned that adding this code may disrupt the readability of the main function. Additionally, it appears that other flags may also be missing their corresponding functions. What do you think? Thanks!

Reply:
Ideally, all flags would have their corresponding functions. But since it's not consistent across this command, I'll leave it up to you to decide:)